datadog_checks_dev/datadog_checks/dev/tooling/commands/meta/snmp/generate_traps_db.py

```diff
@@ -0,0 +1,300 @@
+)
+@click.option(
+    '--output-file',
+    help='Path to a file to store a compacted version of the traps database file. Do not use with --output-dir',
+    type=click.Path(exists=False, dir_okay=False, resolve_path=True),
+)
+@click.option(
+    '--output-format',
+    type=click.Choice(['yaml', 'json'], case_sensitive=False),
+    default='yaml',
+    help='Use json instead of yaml for the output file(s).',
+)
+@click.option('--debug', '-d', help='Include debug output', is_flag=True)
+@click.argument('mib-files', nargs=-1, required=True, type=click.Path(exists=True, dir_okay=False, resolve_path=True))
+def generate_traps_db(mib_sources, output_dir, output_file, output_format, debug, mib_files):
+    """Generate yaml or json formatted documents containing various information about traps. These files can be used by
+    the Datadog Agent to enrich trap data.
+    This command is intended for "Network Devices Monitoring" users who need to enrich traps that are not automatically
+    supported by Datadog.
+
+    The expected workflow is as such:\n
+    1- Identify a type of device that is sending traps that Datadog does not already recognize.\n
+    2- Fetch all the MIBs that Datadog does not support.\n
+    3- Run `ddev meta snmp generate-traps-db -o ./output_dir/ /path/to/my/mib1 /path/to/my/mib2`\n
+    """
+    if debug:
+        set_debug()
+        from pysmi import debug
+
+        debug.setLogger(debug.Debug('all'))
+
+    # Defaulting to github.com/DataDog/mibs.snmplabs.com/
+    mib_sources = [mib_sources] if mib_sources else [MIB_SOURCE_URL]
+
+    if output_file:
+        allowed_extensions = ALLOWED_EXTENSIONS_BY_FORMAT[output_format]
+        if not any(output_file.endswith(x) for x in allowed_extensions):
+            echo_warning(
+                "Output file {} does not end with an allowed extension '{}'".format(
+                    output_file, ", ".join(allowed_extensions)
+                )
+            )
+            output_file = output_file.rsplit('.', 1)[0] + allowed_extensions[0]
+            echo_warning("Using {} instead.".format(output_file))
+
+    with TempDir('ddev_mibs') as compiled_mibs_sources:
+        compiled_mibs_sources = os.path.abspath(compiled_mibs_sources)
+        if output_dir and output_file:
+            abort("Do not set both --output-dir and --output-file at the same time.")
+        elif not output_file and not output_dir:
+            abort("Need to set one of --output-dir or --output-file")
+
+        echo_info("Writing intermediate compiled MIBs to {}".format(compiled_mibs_sources))
+
+        mibs_sources_dir = os.path.join(compiled_mibs_sources, 'mibs_sources')
+        if not os.path.isdir(mibs_sources_dir):
+            os.mkdir(mibs_sources_dir)
+
+        mib_sources = (
+            sorted(set([os.path.abspath(os.path.dirname(x)) for x in mib_files if os.path.sep in x])) + mib_sources
+        )
+
+        mib_files = [os.path.basename(os.path.splitext(x)[0]) for x in mib_files]
```

Comment:
```suggestion
        mib_files = [os.path.basename(x) for x in mib_files]
```
to handle cases where the file have an extension e.g. MY-MIB.txt